### ISSUE TITLE
Document what this actually is

### ISSUE THREAD

**Issue by filips123** [d14na/zerovue#3]
Please document what this actually is. Just repository description and default README aren't enough 😃. Make better README instead.

---

@d14na I also don't know what exactly are you doing here.

Are you doing complete ZeroNet implementation in JS? In this case you should also check [existing ZeroNet prototypes in JS](https://www.npmjs.com/search?q=zeronet) instead of making yet another incompatible instance. You should also check existing issues on main ZeroNet repositories about standardization and protocols as it won't be good if you would create yet another slightly different incompatible ZeroNet instance.

Or are you making just client to existing ZeroFrame WebSocket API? In this case you should check [ZeroFrameJS](https://github.com/filips123/ZeroFrameJS) which is my client for ZeroFrame WebSocket API in JavaScript.

**Comment by d14na** [comment]
tl;dr Although its public, this repo is currently just a "new" prototyping area for code that my team and I have been working on for over a year now. 

---

Hey @filips123, thanks for checking out the repo; and it appears that others have recently taken an interest as well. The attention is more than welcome. To be clear, the main objective of this repo is to expedite the development of https://github.com/d14na/zeronet-explorer, which had to be completely written recently due to new Google Play store restrictions. Afterwards, a [desktop client](1) would be the next objective.

> Are you doing complete ZeroNet implementation in JS?

yes. static zites are complete. dynamic zites are incomplete (still in research & development stage).

> check existing ZeroNet prototypes in JS

yes, of course. already been in direct contact with most zeronet devs.

> check ZeroFramePy

not sure if I've seen that before, but I'll check it out now. ___NOTE: the core codebase is NOT MIT, which makes it unusable/incompatible with any code from our team.___ i'm sure this codebase will be a very useful reference.

> Make better README instead.

absolutely! on the next code push.

**Comment by filips123** [comment]
> not sure if I've seen that before, but I'll check it out now. NOTE: the core codebase is NOT MIT, which makes it unusable/incompatible with any code from our team. i'm sure this codebase will be a very useful reference.

Here do you mean ZeroNet source code, right? My library is licenced under MIT so this should be OK.

@d14na Are you sure? I'm not a lawyer, but I have seen that some people say that it is legal to include GPL2 code in MIT project if it is properly noted and documented with all changes. But some people say that it is not...

Can't you ask @HelloZeroNet about this? As an author of that code, he could know this or add special approvement for your usage.

**Comment by HelloZeroNet** [comment]
The license only matters if you plan to make modification to the main client's source code. If you don't make any modification, then it's compatible with any licenses. (That's how commercial applications uses eg.: OpenSSL)

**Comment by d14na** [comment]
> Here do you mean ZeroNet source code, right? My library is licenced under MIT so this should be OK.

yeah, I haven't checked it out yet, but I will this week. will be back working on the js client for zeronet-explorer.

> Are you sure? I'm not a lawyer

not sure, also not a lawyer.

---

> The license only matters if you plan to make modification to the main client's source code.

that's what I figured too. what concerns me is when your cloning a core library with a new library using a _more permissive license_. not even gonna stress it right now, once the first version of supeer is released, it shouldn't really matter anyway. but again, not a lawyer..
